pkg/client/informers/informers_generated/generic.go

```diff
@@ -0,0 +1,249 @@
+
+package informers_generated
+
+import (
+	"fmt"
+	api "k8s.io/kubernetes/pkg/api"
+	api_v1 "k8s.io/kubernetes/pkg/api/v1"
+	apps "k8s.io/kubernetes/pkg/apis/apps"
+	v1beta1 "k8s.io/kubernetes/pkg/apis/apps/v1beta1"
+	autoscaling "k8s.io/kubernetes/pkg/apis/autoscaling"
+	v1 "k8s.io/kubernetes/pkg/apis/autoscaling/v1"
+	batch "k8s.io/kubernetes/pkg/apis/batch"
+	batch_v1 "k8s.io/kubernetes/pkg/apis/batch/v1"
+	v2alpha1 "k8s.io/kubernetes/pkg/apis/batch/v2alpha1"
+	certificates "k8s.io/kubernetes/pkg/apis/certificates"
+	v1alpha1 "k8s.io/kubernetes/pkg/apis/certificates/v1alpha1"
+	extensions "k8s.io/kubernetes/pkg/apis/extensions"
+	extensions_v1beta1 "k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
+	policy "k8s.io/kubernetes/pkg/apis/policy"
+	policy_v1beta1 "k8s.io/kubernetes/pkg/apis/policy/v1beta1"
+	rbac "k8s.io/kubernetes/pkg/apis/rbac"
+	rbac_v1alpha1 "k8s.io/kubernetes/pkg/apis/rbac/v1alpha1"
+	storage "k8s.io/kubernetes/pkg/apis/storage"
+	storage_v1beta1 "k8s.io/kubernetes/pkg/apis/storage/v1beta1"
+	cache "k8s.io/kubernetes/pkg/client/cache"
+	schema "k8s.io/kubernetes/pkg/runtime/schema"
+)
+
+// GenericInformer is type of SharedIndexInformer which will locate and delegate to other
+// sharedInformers based on type
+type GenericInformer interface {
+	Informer() cache.SharedIndexInformer
+	Lister() cache.GenericLister
+}
+
+type genericInformer struct {
+	informer cache.SharedIndexInformer
+	resource schema.GroupResource
+}
+
+func (f *genericInformer) Informer() cache.SharedIndexInformer {
+	return f.informer
+}
+
+func (f *genericInformer) Lister() cache.GenericLister {
+	return cache.NewGenericLister(f.Informer().GetIndexer(), f.resource)
+}
+
+// ForResource gives generic access to a shared informer of the matching type
+// TODO extend this to unknown resources with a client pool
+func (f *sharedInformerFactory) ForResource(resource schema.GroupResource) (GenericInformer, error) {
+	switch resource {
+	// Group=Apps, Version=InternalVersion
+	case apps.Resource("statefulsets"):
+		return &genericInformer{resource: resource, informer: f.Apps().InternalVersion().StatefulSets().Informer()}, nil
+
+		// Group=Apps, Version=V1beta1
```

Comment:
You are right. I double checked the client-gen code, it uses the full group name [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/clientset_generated/release_1_5/typed/authentication/v1beta1/authentication_client.go#L71), and it's parsed from a tag [here](https://github.com/kubernetes/kubernetes/blob/master/cmd/libs/go2idl/client-gen/generators/generator_for_group.go#L83-L85), the tag is defined [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/authentication/v1beta1/doc.go#L19). The mechanism looks quite hacky though.